### ISSUE TITLE
Specify directory >= 1.2.2 to avoid dangerous removeDirectoryRecursive behavior in withTempDirectory

### ISSUE THREAD

**Issue by ddssff** [batterseapower/temporary#6]
The `removeDirectoryRecusive` function will follow symbolic links if they point to directories (due to the behavior of `doesDirectoryExist`.)  This could conceivably lead to files outside the temporary directory being removed when `withTemporaryDirectory` finishes.

**Comment by feuerbach** [comment]
I tried to use the function in the way you describe, and didn't observe the dangerous behavior. Also, [the docs for `removeDirectoryRecursive`](https://hackage.haskell.org/package/directory-1.2.5.0/docs/System-Directory.html#v:removeDirectoryRecursive) say: "Within this directory, symbolic links are removed without affecting their the targets."

**Comment by ddssff** [comment]
This, or at least the documentation, seems to have changed as of directory-1.2.2.0:

https://hackage.haskell.org/package/directory-1.2.2.0/docs/System-Directory.html#v:removeDirectoryRecursive

https://hackage.haskell.org/package/directory-1.2.1.0/docs/System-Directory.html#v:removeDirectoryRecursive

**Comment by ddssff** [comment]
...which was uploaded in March of 2015...

**Comment by ddssff** [comment]
It looks like 1.2.1.0 actually did have the dangerous behavior:

https://hackage.haskell.org/package/directory-1.2.1.0/docs/src/System-Directory.html#removeDirectoryRecursive

**Comment by feuerbach** [comment]
Good catch! I don't think I should impose a constraint on directory, but I'll certainly add a mention to the documentation.
